runtime/jvmti/jvmtiHelpers.c

```diff
@@ -174,13 +175,14 @@ disposeEnvironment(J9JVMTIEnv * j9env, UDATA freeData)
 #if JAVA_SPEC_VERSION >= 11
 		if (j9env->capabilities.can_generate_sampled_object_alloc_events) {
 			J9JVMTI_DATA_FROM_VM(vm)->flags &= ~J9JVMTI_FLAG_SAMPLED_OBJECT_ALLOC_ENABLED;
+			/* Set sampling interval to UDATA_MAX to inform GC that sampling is not required */
+			vm->memoryManagerFunctions->j9gc_set_allocation_sampling_interval(currentThread, UDATA_MAX);
```

Comment:
Do we need some way to push this out to all the threads? (ie: an async event?) Is this event per thread or global?

Reply:
If I understand the question correctly, there is no need to do so for all threads.
`j9gc_set_allocation_sampling_interval(vmThread, samplingInterval)` modifies `MM_GCExtensions.oolObjectSamplingBytesGranularity`
https://github.com/eclipse/omr/blob/eb71465140b3b7fd74f8ad12a65da4d384aa945e/gc/base/GCExtensionsBase.hpp#L295
which is a global setting for GC to determine if the newly added `MM_GCExtensions. _traceAllocationBytes` (https://github.com/eclipse/omr/pull/5071/files) has passed the `samplingInterval` and a sampling event should be fired.

`can_generate_sampled_object_alloc_events` is global, the sampled objects come from the allocations in different threads.